### DISCUSSION TITLE
how to run docker compose without docker desktop

### DISCUSSION THREAD

**Discussion by msh2050** [WasmEdge/WasmEdge#2558] [General]
I tried to build with docker engine inside WSL, but I get platform is not supported!
I see in the documentation the docker CLI method is coming soon 

any hint please

**Comment by juntao** [comment]
Hi

You will need 24.0.0-beta-1 release of dockerd. You can run wasm workloads with this build (you do have to activate the containerd feature, put `{ "features": [ "containerd-snapshotter": true ] }` in `/etc/docker/daemon.json`). 

You can download the build from 

http://download.docker.com 

For debian for example it’s here: 

https://download.docker.com/linux/debian/dists/bullseye/pool/test/amd64/docker-ce_24.0.0~beta.1-1~debian.11~bullseye_amd64.deb

**Comment by msh2050** [comment]
I installed the beta version but I get this error

`docker: Error response from daemon: failed to create task for container: failed to start shim: failed to resolve runtime path: runtime "io.containerd.wasmedge.v1" binary not installed "containerd-shim-wasmedge-v1": file does not exist: unknown.`

this id my docker info:
```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.17.2
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.23.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 24.0.0-beta.1
 Storage Driver: overlayfs
  driver-type: io.containerd.snapshotter.v1
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 6.1.21.2-microsoft-standard-WSL2
 Operating System: Ubuntu 22.04.2 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.827GiB
 Name: *****
 ID: *****-****-****-*****-*************
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  *****/8
 Live Restore Enabled: false
```
